content/posts/ai-crawlers-abuse.md

```diff
@@ -0,0 +1,177 @@
+title: AI Crawlers need to be more respectful
+date: 2024-07-25
+category: Engineering
+tags: ai, crawlers, abuse
+authors: Eric Holscher
+status: published
+image:
+image_credit:
+
+Many AI companies are starting to crawl many pages on the web,
+trying to extract as much information as fast as possible.
+It feels like a new gold rush,
+and in their haste,
+some of these crawlers are being so badly they are causing harm to the sites they are crawling.
+
+At Read the Docs,
+we host documentation for many projects and are generally bot friendly,
+but the behavior of AI crawlers is currently causing us problems.
+We host HTML zip files (``htmlzip``) of documentation for many projects,
+and these can be somewhat large.
+The AI Crawlers are downloading these files repeatedly,
+and without any rate or bandwidth limiting.
+
+AI crawlers are acting in a way that is not respectful to the sites they are crawling,
+and that is going to cause a backlash against AI crawlers in general.
+As a community-supported site without a large budget,
+AI crawlers have cost us a significant amount of money in bandwidth charges,
+and caused us to spend a large amount of time dealing with abuse.
+
+## AI crawler abuse
+
+I want to start off with a few examples of the types of abuse we are seeing:
+
+### 73 TB in March 2024 from one crawler
+
+**One crawler downloaded 73 TB of htmlzip files in May 2024, with almost 10 TB in a single day**. This cost us over $5,000 in bandwidth charges, and we had to block the crawler. We emailed this company, reporting a bug in their crawler, and we're working on them reimbursing us for the costs.
```

Comment:
I mentioned above we explained the problem well down here, but it might be worth moving the description of these files here. I would definitely mention these files, you mention below they don't change often too, which maybe helps with context here.

Reply:
```suggestion
**One crawler downloaded 73 TB of htmlzip files in May 2024, with almost 10 TB in a single day**. This cost us over $5,000 in bandwidth charges, and we had to block the crawler. We emailed this company, reporting a bug in their crawler, and we're working with them on reimbursing us for the costs.
```